### PR TITLE
Aerogear 2521

### PR DESCRIPTION
**Pull request by StephenCoady**
## Motivation

Merge existing security app features to the showcase application.

JIRA: https://issues.jboss.org/browse/AEROGEAR-2521

## Description

Bring in Access Control, device trust, network and secure storage. Also create a sub menu so that security features do not make the menubar cluttered in the application.

## Progress

- [x] Access control
- [x] Device trust
- [x] Network
- [x] Sub menu implemented
- [x] Secure storage

## Verification 

* From within this repo run the following:
```
npm install
npm link @aerogear/auth
npm run ionic:android
npm run ionic:ios
```
* For both platforms ensure that all pages work and functionality is the same as the existing security app.

### CODE REVIEW

**Comment by StephenCoady** [comment]
circleci is failing because of this PR's reliance on new function added to the SDK.

File: config.xml [hunk 1]
```diff
@@ -80,6 +80,10 @@
     <plugin name="cordova-plugin-device" spec="^2.0.1" />
     <plugin name="cordova-plugin-splashscreen" spec="^5.0.2" />
     <plugin name="cordova-plugin-ionic-webview" spec="^1.2.0" />
-    <plugin name="cordova-plugin-ionic-keyboard" spec="^2.0.5" />
     <plugin name="cordova-plugin-inappbrowser" spec="~3.0.0" />
+    <allow-navigation href="http://10.201.82.242:8100" />
+    <plugin name="cordova-plugin-iroot" spec="^0.7.0" />
```
Comment (wtrocki) [inline config.xml]: Quick question. Are we planning to wrap those plugins at some point to reflect what we done with Core (and Auth in future)?

Comment (wtrocki) [reply config.xml]: @cianclarke Cordova JS provides now means for this usage and we spiked this already. Let's sync on this on the other PR.

**Comment by wtrocki** [comment]
Really happy with this change. Verified locally.

**Comment by jhellar** [comment]
Tested on iOS emulator - auth still works fine. I have not tested security app features properly as they were not tested on iOS before.

**Comment by ciaranRoche** [comment]
@StephenCoady are we ok to merge this? It will be good to have this merged as we progress with the trust checks!

**Comment by StephenCoady** [comment]
@ciaranRoche I think it is ok to merge. I have ran through all pages on both android and iOS and everything seems to be working.

File: src/pages/security/network/network.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  apiServerUrl: string;
+  apiEndpoint: string;
+  pinningSuccess: boolean;
+  responseRecieved: boolean;
+  requestSuccess: boolean;
+  requestFailure: boolean;
+  configuration: any;
+
+  constructor(public loadingCtrl: LoadingController, public toastCtrl: ToastController, private auth: AuthService, public navCtrl: NavController, public http: Http) {
+    this.configuration = new ConfigurationHelper(config).getConfig(AuthService.ID);
```
Comment (wtrocki) [inline src/pages/security/network/network.ts]: Just noticed that we using this. Is that temporary?

Comment (wtrocki) [inline src/pages/security/network/network.ts]: This element is internal part of the SDK. We going to remove it from imports now.

Comment (wtrocki) [inline src/pages/security/network/network.ts]: @StephenCoady Just got fresh master. This element is internal part of the SDK. We going to remove it from imports now. Is that change temporary?

Comment (StephenCoady) [reply src/pages/security/network/network.ts]: @wtrocki yes you are right. this shouldn't have been included here. I will add it to the auth module and consume from there. great catch, thanks!

**Pull request merged**